starter-code/index.js

```diff
@@ -1,11 +1,75 @@
 class SortedList {
-    constructor() {}
-    add(item) {}
-    get(pos) {}
-    max() {}
-    min() {}
-    average() {}
-    sum() {}
+    constructor() {
+      this.items =[];
+      this.length = 0;
+    }
+    add(item) {
+      this.items.push(item);
+      this.items.sort(function(a, b) {
+        return a - b;
+      });
+      this.length ++;
+    }
+    get(pos) {
+      if(pos > this.items.length) {
+        throw new Error("OutOfBounds")
+      }
+      else {
+        return this.items[pos];
+      }
+    }
+    max() {
+      if(this.items.length === 0) {
+        throw new Error("EmptySortedList")
+      }
+      else {
+        var max = 0;
+        this.items.forEach(function(item) {
+          if(item > max) {
+            max = item;
+          }
+        })
+        return max
+      }
+    }
+    min() {
+      if(this.items.length === 0) {
+        throw new Error("EmptySortedList")
+      }
+      else {
+        var min = 100;
+        this.items.forEach(function(item) {
+          if(item < min) {
+            min = item;
+          }
+        })
+        return min
+      }
+    }
+    average() {
+      if(this.items.length === 0) {
+        throw new Error("EmptySortedList")
+      }
+      else{
+        var sum = this.items.reduce(function(acc , cur) {
```

Comment:
Au lieu de décrire ta future fonction sum, tu pouvais l'appeler directement ici (même si tu la définis après) : 
average() {
    if (this.length <= 0) {
      throw new Error("EmptySortedList");
    }
    return this.sum(this.items) / this.length;
  }